plugins/annotations.md

```diff
@@ -0,0 +1,227 @@
+*
+* @method annotationChanged
+* @param {String} name Name of annotation to listen for
+* @param {function} callback Calback with (state, name, and data) fired when the annotation
+* at the cursor changes. If state if false, data will not be provided.
+*/
+annotationChanged: (name: string, callback): void
+```
+## Example
+
+To create the Annotate API, use the following example:
+
+```js
+<script type="text/javascript">
+tinymce.init({
+   selector: "textarea",
+   plugins: [
+       "advlist autolink lists link image charmap print preview anchor",
+       "searchreplace visualblocks code fullscreen",
+       "insertdatetime media table contextmenu paste"
+   ],
+   toolbar: "annotate-alpha | insertfile undo redo | styleselect | bold italic | alignleft aligncenter alignright alignjustify | bullist numlist outdent indent | link image",
+
+   content_style: '.mce-annotation { background-color: darkgreen; color: white; }',
+
+   setup: (ed) => {
+     ed.addButton('annotate-alpha', {
+       text: 'Annotate',
+       onclick: () => {
+         const comment = prompt('Comment with?');
+         ed.experimental.annotator.annotate('alpha', {
+           comment
+         });
+         ed.focus();
+       },
+
+       onpostrender: (ctrl) => {
+         const button = ctrl.control;
+         ed.on('init', () => {
+           ed.experimental.annotator.annotationChanged('alpha', (state, name, obj) => {
+             if (! state) {
+               button.active(false);
+             } else {
+               button.active(true);
+             }
+           });
+         });
+       }
+     });
+
+     ed.on('init', () => {
+       ed.experimental.annotator.register('alpha', {
+         persistent: true,
+         decorate: (uid, data) => {
+           return {
+             attributes: {
+               'data-mce-alpha': data.alpha
+             }
+           };
+         }
+       });
+
+     });
+   }
+});
+
+</script>
+
+<form method="post" action="dump.php">
+    <textarea name="content"></textarea>
+</form>
+```
+[Example]({{ site.baseurl }}/images/annotate.png)
+
+## Retrieving All Annotations for a Particular Annotation Name
+
+Lastly, the annotator API allows you to retrieve an object of all of the uids for a particular annotation type (e.g. alpha), and the nodes associated with those uids. For example, to retrieve all `alpha` annotations, we would use this code:
+
+```js
+var annotations = editor.annotator.getAll('alpha');
+var nodesInFirstUid = annotations['first-uid'];
+```
+
+Assuming that there is a uid called `first-uid`, the above code shows you how to access the nodes used for making that annotation. The full API is:
+
+```js
+/**
+* Retrieve all the annotations for a given name
+*
+* @method getAll
+* @param {String} name the name of the annotations to retrieve
+* @return {Object} an index of annotations from uid => DOM nodes
+*/
+getAll: (name: string): Record<string, DomElement[]>
```

Comment:
This appears to be TypeScript. Should we be using TypeScript or JavaScript here?